### PR TITLE
Fix `if not disabler()` case

### PR DESCRIPTION
**Pull request by xclaesse**
Closes: #6010

### CODE REVIEW

File: test cases/common/163 disabler/meson.build [hunk 1]
```diff
@@ -76,3 +76,25 @@ else
...
 endif
 assert(has_not_changed, 'App has changed.')
+
+if_is_disabled = true
+if disabler()
+  if_is_disabled = false
+else
+  if_is_disabled = false
+endif
+assert(if_is_disabled, 'Disabler in "if condition" must skip both blocks')
```
Comment (emersion) [inline test cases/common/163 disabler/meson.build]: Should it skip both blocks? I'd expect this code to print "missing header":

```meson
if cc.has_header('libavutil/hwcontext_drm.h', dependencies: libavutil):
    warning('has header')
else
    warning('missing header')
endif
```

Comment (xclaesse) [reply test cases/common/163 disabler/meson.build]: no, it skip both blocks. A disabler() is not a not-found dependency object.

Comment (xclaesse) [reply test cases/common/163 disabler/meson.build]: It is really a special type of object, unlike anything I've seen in other languages. You have understand how it works to use it, you could have surprises ;-)

**Comment by emersion** [comment]
I can confirm this fixes our build: https://builds.sr.ht/~emersion/job/95288#task-wlroots

**Comment by xclaesse** [comment]
@jpakkane while at this, I added a test for disabler in foreach loop. Currently it does nothing special, it sets the iterator to a disabler() object and execute the block. IMHO that's a sensible behaviour, but we could also just skip the code block when the iterator is a disabler (that would be behaviour change), or even skip the whole foreach loop if only one item is a disabler (I wouldn't recommend that)?

**Comment by jpakkane** [comment]
An alternative approach would be that putting a disabler in an array or dict makes the containing object (array or dict) into a disabler. Not sure what the reasonable thing to do here would be.

It is not a decision for this MR, though, so merging.

**Comment by scivision** [comment]
This code is very helpful and useful, thank you.

**Pull request merged**